subgraphs/diva-staking/utils/tranches.ts

```diff
@@ -97,99 +109,64 @@ export function getRedemptionTranchesForAllDeposits(
-    let currentTrancheAmount = deposit.trancheAmounts[i];
+    let depositTrancheAmount = useTrancheAmount(deposit.trancheAmounts[i]);
 
     // skip tranches without money deposited to
-    if (currentTrancheAmount.equals(ZERO_BD)) {
+    if (depositTrancheAmount.amount.equals(ZERO_BD)) {
       continue;
     }
 
-    if (currentTrancheAmount >= amountLeftToRedeem) {
-      tranchesRedeemedFrom.push(new Tranche(amountLeftToRedeem, i));
+    if (depositTrancheAmount.amount >= amountLeftToRedeem) {
+      redemptionTrancheAmounts.push(
+        createTrancheAmount(i, amountLeftToRedeem, depositTrancheAmount.startStakingAt, redemptionTimestamp, event),
+      );
+
+      depositTrancheAmount.amount = depositTrancheAmount.amount.minus(amountLeftToRedeem);
+      depositTrancheAmount.save();
+
       amountLeftToRedeem = ZERO_BD;
       break; // we have redeemed all the funds
     } else {
-      tranchesRedeemedFrom.push(new Tranche(currentTrancheAmount, i));
-      amountLeftToRedeem = amountLeftToRedeem.minus(currentTrancheAmount);
-    }
-  }
-
-  return new RedemptionTranchesForDepositResponse(tranchesRedeemedFrom, amountLeftToRedeem, deposit);
-}
-
-export function getAggregatedRedemptionTranches(
-  redemptionTranchesForAllDeposits: RedemptionTranchesForDepositResponse[],
-): Tranche[] {
-  let tranches: Tranche[] = [];
-
-  for (let i = 0; i < redemptionTranchesForAllDeposits.length; i++) {
-    let redemptionTranchesForSingleDeposit = redemptionTranchesForAllDeposits[i].tranches;
-
-    for (let j = 0; j < redemptionTranchesForSingleDeposit.length; j++) {
-      let singleRedemptionTrancheForSingleDeposit = redemptionTranchesForSingleDeposit[j];
-
-      // check if tranche already exists
-      let trancheAlreadyExists = false;
-      for (let k = 0; k < tranches.length; k++) {
-        let tranche = tranches[k];
-
-        if (tranche.id === singleRedemptionTrancheForSingleDeposit.id) {
-          tranches[k].amount = tranches[k].amount.plus(singleRedemptionTrancheForSingleDeposit.amount);
-          trancheAlreadyExists = true;
-          break;
-        }
-      }
-
-      if (trancheAlreadyExists == false) {
-        // even though redemptionTrancheForDeposit is also Tranche type we create new Tranche class in order not to have reference to old Tranche,
-        // which could result in bug when we modify tranche
-        tranches.push(
-          new Tranche(singleRedemptionTrancheForSingleDeposit.amount, singleRedemptionTrancheForSingleDeposit.id),
-        );
-      }
+      redemptionTrancheAmounts.push(
+        createTrancheAmount(
+          i,
+          depositTrancheAmount.amount,
+          depositTrancheAmount.startStakingAt,
+          redemptionTimestamp,
+          event,
+        ),
+      );
+
+      depositTrancheAmount.amount = ZERO_BD;
+      depositTrancheAmount.save();
+
+      amountLeftToRedeem = amountLeftToRedeem.minus(depositTrancheAmount.amount);
```

Comment:
here it will be a bug because `depositTrancheAmount.amount` is set to 0 above. We could move it above the line `depositTrancheAmount.amount = ZERO_BD;`

Reply:
Good spot! Fixed.